### ISSUE TITLE
index drawn by AliasMethod is not on the same gpu as the model

### ISSUE THREAD

**Issue by dongyaoli10x** [HobbitLong/CMC#58]
not sure if I missed something but it seems to me that if you train on multiple gpus with current implementation, the `AliasMethod`  puts the index on default gpu. The `memory_l` and `memory_ab` are on the correct gpu using the `register_buffer`. Then the `torch.index_select(self.memory_l, 0, idx.view(-1)).detach()` would gives `arguments are located on different GPUs ` error.
